pi-apps-terminal-bash-edition.sh

```diff
@@ -148,106 +90,154 @@ while [ "$1" != "" ]; do
 		;;
 		install)
 			shift
-			for arg in "$@"; do
-				cmdflags+="$arg "
-			done
-			$PI_APPS_DIR/manage install "$(echo $cmdflags)"
-			exit $?
+			if [ -d "$DIRECTORY/apps/$*" ]; then #if only provides one app
+				for arg in "$@"; do
+					cmdflags+="$arg "
+				done
+				"$DIRECTORY/manage" install "$cmdflags"
+				exit $?
+			else #multiple app
+				for arg in "$@"; do
+					cmdflags+="$arg\n"
+				done
+				#remove last \'n'
+				args=${cmdflags%\\n}
+				#install apps
+				"$DIRECTORY/manage" multi-install "$(echo -e "$args")"
+				exit $?
+			fi
 		;;
-		multi-install)
-			shift
-			for arg in "$@"; do
-				cmdflags+="$arg\n"
-			done
-			#remove last \'n'
-			args=${cmdflags%\\n}
-			#install apps
-			$PI_APPS_DIR/manage multi-install "$(echo -e "$args")"
-			exit $?
-			;;
 		remove|uninstall)
 			shift
-			for arg in "$@"; do
-				cmdflags+="$arg "
-			done
-			$PI_APPS_DIR/manage uninstall "$(echo $cmdflags)"
-			exit $?
+			if [ -d "$DIRECTORY/apps/$*" ]; then #if only provides one app
+				for arg in "$@"; do
+					cmdflags+="$arg "
+				done
+				"$DIRECTORY/manage" uninstall "$cmdflags"
+				exit $?
+			else #multiple app
+				for arg in "$@"; do
+					cmdflags+="$arg\n"
+				done
+				#remove last \'n'
+				args=${cmdflags%\\n}
+				#uninstall apps
+				"$DIRECTORY/manage" multi-uninstall "$(echo -e "$args")"
+				exit $?
+			fi
 		;;
-		multi-remove | multi-uninstall)
-			shift
-			for arg in "$@"; do
-				cmdflags+="$arg\n"
-			done
-			args=${cmdflags%\\n}
-			#uninstall apps
-			$PI_APPS_DIR/manage multi-uninstall "$(echo -e "$args")"
-			exit $?
-			;;
 		reinstall)
 			shift
-			for arg in "$@"; do
-				cmdflags+="$arg "
-			done
-			cmdflags="${cmdflags::-1}"
-			$PI_APPS_DIR/manage uninstall "$cmdflags"
-			$PI_APPS_DIR/manage install "$cmdflags" || error "Failed to reinstall \"$cmdflags\"!"
-			exit $?
+			if [ -d "$DIRECTORY/apps/$*" ]; then #if only provides one app
+				for arg in "$@"; do
+					cmdflags+="$arg "
+				done
+				"$DIRECTORY/manage" uninstall "$cmdflags"
+				"$DIRECTORY/manage" install "$cmdflags"
+				exit $?
+			else #multiple app
+				for arg in "$@"; do
+					cmdflags+="$arg\n"
+				done
+				#remove last \'n'
+				args=${cmdflags%\\n}
+				"$DIRECTORY/manage" multi-uninstall "$(echo -e "$args")"
+				"$DIRECTORY/manage" multi-install "$(echo -e "$args")"
+				exit $?
+			fi
 		;;
 		list-installed)
 			#list all the installed apps
-			#list_apps installed
-			ls "$PI_APPS_DIR/apps" | GREP_COLORS='ms=1;34' grep --color=always -x "$(grep -rx 'installed' "${PI_APPS_DIR}/data/status" | awk -F: '{print $1}' | sed 's!.*/!!' | sed -z 's/\n/\\|/g' | sed -z 's/\\|$/\n/g')"
+			list_apps installed
```

Comment:
```suggestion
			#list_apps installed
			ls "$PI_APPS_DIR/apps" | GREP_COLORS='ms=1;34' grep --color=always -x "$(grep -rx 'installed' "${PI_APPS_DIR}/data/status" | awk -F: '{print $1}' | sed 's!.*/!!' | sed -z 's/\n/\\|/g' | sed -z 's/\\|$/\n/g')"
```
I used the modified version of `list_apps installed` from the `api` script to get colored output.
so unless it does it by default now (I don't think so) I don't want to change it.